api/capi/src/nnstreamer-capi-util.c

```diff
@@ -868,8 +868,27 @@ ml_validate_model_file (const char *model, ml_nnfw_type_e * nnfw)
   switch (*nnfw) {
     case ML_NNFW_TYPE_ANY:
       if (g_str_has_suffix (path_down, ".tflite")) {
-        ml_logi ("The given model [%s] is supposed a tensorflow-lite model.", model);
-        *nnfw = ML_NNFW_TYPE_TENSORFLOW_LITE;
+        /**
+         * .tflite is supported by both tensorflow and nnfw.
+         * Priority decided with ini file.
+         */
+        gboolean nnfw_runtime_priority = nnsconf_get_custom_value_bool (
+            "nnfw-runtime", "prioritize_tflite_ext", FALSE);
+        bool available_nnfw = FALSE, available_tflite = FALSE;
+
+        ml_check_nnfw_availability (ML_NNFW_TYPE_NNFW, ML_NNFW_HW_ANY,
+            &available_nnfw);
+        ml_check_nnfw_availability (ML_NNFW_TYPE_TENSORFLOW_LITE,
+            ML_NNFW_HW_ANY, &available_tflite);
+
+        if ((nnfw_runtime_priority && available_nnfw) ||
+            (!nnfw_runtime_priority && !available_tflite)) {
+          ml_logi ("The given model [%s] is supposed a nnfw model.", model);
+          *nnfw = ML_NNFW_TYPE_NNFW;
```

Comment:
What if ```!available_tflite && !available_nnfw)```?

Reply:
In this function case ML_NNFW_TYPE_ANY, the purpose is to assign a **possible** valid NN framework. Checking for availability is an added bonus I think. 
My assumption is `ml_validate_model_file` is followed by a call to `ml_check_nnfw_availability`.
The case of `!available_tflite && !available_nnfw` will be covered by that.

So, to make this better, I have added a check of `ml_check_nnfw_availibility` at the end of `ML_NNFW_TYPE_ANY` in `ml_validate_model_file` to check if the assigned nnfw is even available or not for all the ext. This covers the case of `!available_tflite && !available_nnfw`, and for other cases of `ML_NNFW_TYPE_ANY`.